### DISCUSSION TITLE
Latency when sending BLE Midi on ESP32

### DISCUSSION THREAD

**Discussion by jhsa** [tttapa/Control-Surface#647] [General]
I am experiencing a few milliseconds Latency when sending  MIDI over BLE MIDI on a ESP32 module. Is that normal?
I use an app for my backing tracks. Every song has an automation track where I program the MIDI information. There is no Delay when sending over USB Midi.
Thanks
João

**Comment by tttapa** [comment]
You'll always have more delay when using Bluetooth, but you can try to minimize it using the [`BluetoothMIDI_Interface::setTimeout()`](https://tttapa.github.io/Control-Surface-doc/Doxygen/d0/d89/classBluetoothMIDI__Interface.html#a28396bedf9f29fa7bfd0b893a92f8718) method. The default timeout is 10ms.

**Reply by jhsa** [reply]
Thank you. Will check it.

João

**Comment by KingMichaelCoy** [comment]
If I want to build something like

[](https://learn.adafruit.com/web-ble-midi/ble-midi-controller)

But for esp32 how could I do that?
